File: src/lua/rendering/Rectangle.lua
local Color = require("rendering.Color")

local Rectangle = {x = 0, y = 0, width = 1, height = 1, color = Color:White()}

function Rectangle:New(x, y, width, height, color)
    local o = {
        x = x or self.x,
        y = y or self.y,
        width = width or self.width,
        height = height or self.height,
        color = color or self.color
    }
    setmetatable(o, self)
    self.__index = self
    return o
end

function Rectangle:SetX(x)
    self.x = x
end

function Rectangle:SetY(y)
    self.y = y
end

function Rectangle:SetWidth(width)
    self.width = width
end

function Rectangle:SetHeight(height)
    self.height = height
end

function Rectangle:Draw()
    DrawRectangle(self.x, self.y, self.width, self.height, self.color)
end

return Rectangle

File: src/Renderer/Shapes/Rectangle.cpp
#include <Renderer/Shapes/Rectangle.hpp>

namespace Aloha
{
    Rectangle::Rectangle(int x, int y, int width, int height, Color color)
        : x(x), y(y), width(width), height(height), color(color) {}
    // Rectangle::~Rectangle() {}

    void Rectangle::draw()
    {
        Aloha::drawRectangle(x, y, width, height, color);
    }

    void Rectangle::setX(int newX)
    {
        x = newX;
    }

    void Rectangle::setY(int newY)
    {
        y = newY;
    }

    void Rectangle::setTotalWidth(int newWidth)
    {
        width = newWidth;
    }

    void Rectangle::setHeight(int newHeight)
    {
        height = newHeight;
    }

    int Rectangle::getX()
    {
        return x;
    }

    int Rectangle::getY()
    {
        return y;
    }

    int Rectangle::getWidth()
    {
        return width;
    }

    int Rectangle::getHeight()
    {
        return height;
    }
}